packages/alfa-aria/test/name.spec.tsx

```diff
@@ -961,3 +961,35 @@ test(`.from() correctly handles chained aria-labelledby references`, (t) => {
     ],
   ]);
 });
+
+test(".from() ignores nodes that are not exposed when computing name from content", (t) => {
+  const text = h.text("Hello ");
+  const exposed = <span>{text}</span>;
+  const notExposed = <span aria-hidden="true">World!</span>;
+
+  const heading = (
+    <h1>
+      {exposed}
+      {notExposed}
+    </h1>
+  );
+
+  t.deepEqual(Name.from(heading, device).toArray(), [
+    [
+      Option.of(
+        Name.of("Hello", [
+          Name.Source.descendant(
+            heading,
+            Name.of("Hello", [
+              Name.Source.descendant(
+                exposed,
+                Name.of("Hello ", [Name.Source.data(text)])
```

Comment:
Interestingly, https://w3c.github.io/accname/#step2G does not require trimming of flattening the string, which means that accessible name of text nodes is not a flat string, contrary to what https://w3c.github.io/accname/#mapping_additional_nd_te claims:
> The purpose of the computation is to create a perceivable label or description for alternative presentations, in the form of a flat string of space separated textual tokens.

🤷

Reply:
https://github.com/w3c/accname/issues/63 🤣